### ISSUE TITLE
Current pypi version is "0+unknown"

### ISSUE THREAD

**Issue by rwarren** [python-quantities/python-quantities#122]
Here is what `pip install quantities` currently yields:

```
$ pip install quantities
Collecting quantities
Installing collected packages: quantities
Successfully installed quantities-0+unknown
$ pip freeze | grep quantities
quantities==0+unknown
```

Seems like the weird `0+unknown` thing may be related to versioneer? See [here](https://github.com/prjemian/punx/issues/24), for example.

**Comment by bjodah** [comment]
I have had problems with this too see #109

**Comment by prjemian** [comment]
My resolution to this versioneer problem has been to code my own git-specific solution (https://github.com/prjemian/punx/issues/24).  Not sure if this will work for others yet but trying it on four projects as a trial.

**Comment by swharden** [comment]
This issue persists with me using pip in windows. It installs okay (mild error visible), but gives annoying highlighted error messages any time I try to install or upgrade another pypi package that claims quantities as a dependency. Is there something I can do from the command line (or even from my own setuptools) to prevent the end user from seeing these errors when they simply `pip install` things?

![untitled](https://cloud.githubusercontent.com/assets/4165489/20548335/a464d858-b0f0-11e6-9aab-8911bd234b91.png)

![untitled2](https://cloud.githubusercontent.com/assets/4165489/20548351/be463fb4-b0f0-11e6-8445-23079b7b07c3.png)

**Comment by swharden** [comment]
... FYI I ended up just telling it to install quantities 0.10.1 (instead of 0.11.1) and it worked fine.
`pip install quantities-0.10.1`
or
`pip install quantities-0.10.1 --force-reinstall`

**Comment by rgerkin** [comment]
+1 on this issue.  Having trouble with other packages requiring a specific minimum version, installing this one, and then failing because of the version number.

**Comment by swharden** [comment]
yes my "solution" only works until I pip install --upgrade something, then it tries to upgrade everything, and installs the same version-broken quantities causing this problem in the first place. I really hope they fix this soon...

**Comment by ddale** [comment]
Sorry all, I've had some big life changes (changing job, additions to family, and until this week, no personal computer at home) that has interfered with my ability to maintain this project. I'd be interested in adding new people to the project to help.

So what is the deal with pip and versioneer? I just tried creating a new sdist from my dirty master, and the version reports fine if I run from that directory or install using `python setup.py install`, but if I do a pip install it reports `0+unknown`.

**Comment by prjemian** [comment]
see: https://github.com/python-quantities/python-quantities/issues/109 (2016-04-21)
and: https://github.com/python-quantities/python-quantities/issues/102 (2016-01-08)
and: https://github.com/python-quantities/python-quantities/issues/122 (2016-09-19)

got tired of this as unresolved versioneer problem with this direct solution:
https://github.com/prjemian/spec2nexus/commit/9d8e9ff34f0b916e8f5398cb6b14f2f0d4ed33d6

also reimplemented basic goal of versioneer here:
https://github.com/prjemian/punx/blob/master/src/punx/_version.py

**Comment by ddale** [comment]
Thanks @prjemian, I'll try your approach.

**Comment by bjodah** [comment]
I'd be willing to help out with the project, since I use it a lot myself.
Regarding `__version__` in distributions, FWIW, I have also converged to manually calling `git describe` from `setup.py`.

**Comment by rgerkin** [comment]
Just do [this](https://github.com/scidash/python-quantities/commit/b40da6ccd2e028ca887555ee62e77f07499f98bc) (hard-coded version string) until you have time to fix versioneer or whatever auto-versioning solution you want to use.  You may want to change the version number I provided.

**Comment by apdavison** [comment]
I believe I have fixed the versioneer problem in 4a9d2680485164b055a5f455d8ec14f9ce223f2a

Essentially, `setup.py` needs to get the `sdist` command from versioneer and not directly from distutils
